dds/DCPS/DomainParticipantImpl.cpp

```diff
@@ -623,24 +623,31 @@ DomainParticipantImpl::delete_topic_i(
       CORBA::String_var topic_name = the_topic_servant->get_name();
       TopicMap::mapped_type* entry = 0;
 
-      if (Util::find(topics_, topic_name.in(), entry) == -1) {
+      TopicMapIteratorPair iters = topics_.equal_range(topic_name.in());
+      TopicMapIterator iter;
+      for (iter = iters.first; iter != iters.second; ++iter) {
+        if (iter->second.pair_.svt_ == the_topic_servant) {
+          entry = &iter->second;
+          break;
+        }
+      }
+      if (entry == 0) {
         if (DCPS_debug_level > 0) {
           ACE_ERROR((LM_ERROR,
-                     ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::delete_topic_i, ")
-                     ACE_TEXT("%p\n"),
-                     ACE_TEXT("find")));
+                     ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::delete_topic_i, not found\n")));
```

Comment:
Any logging of id/name helpful?

Reply:
```suggestion
                     ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::delete_topic_i() - topic '%s' not found\n"), topic_name.in()));
```

Reply:
topic_name is char*, so %C

Reply:
Ah, sorry, I saw CORBA so I assumed ACE_TCHAR.